Cannot find module '@kui-shell/client/config.d/autoplay.json' or its corresponding type declarations.

Cloning and running `npm ci` fresh, these errors appear:

    plugins/plugin-client-notebooks/src/index.tsx:27:22 - error TS2307: Cannot find module '@kui-shell/client/config.d/autoplay.json' or its corresponding type declarations.

    27 import autoplay from '@kui-shell/client/config.d/autoplay.json'
                            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

    plugins/plugin-client-notebooks/src/preload.ts:17:22 - error TS2307: Cannot find module '@kui-shell/client/config.d/autoplay.json' or its corresponding type declarations.

    17 import autoplay from '@kui-shell/client/config.d/autoplay.json'
                            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

There's no file `autoplay.json` in `plugins/plugin-client-default/config.d`

--- a/package.json
+++ b/package.json
@@ -33,10 +33,10 @@
   "author": "@starpit",
   "license": "Apache-2.0",
   "devDependencies": {
-    "@kui-shell/builder": "^10.7.6",
-    "@kui-shell/proxy": "^10.7.6",
-    "@kui-shell/react": "^10.7.6",
-    "@kui-shell/webpack": "^10.7.6",
+    "@kui-shell/builder": "^10.8.1",
+    "@kui-shell/proxy": "^10.8.1",
+    "@kui-shell/react": "^10.8.1",
+    "@kui-shell/webpack": "^10.8.1",
     "@types/node": "14.11.8",
     "@types/react": "17.0.20",
     "@types/react-dom": "17.0.9",
@@ -46,16 +46,16 @@
   },
   "dependencies": {
     "@kui-shell/client": "file:./plugins/plugin-client-default",
-    "@kui-shell/core": "^10.7.6",
-    "@kui-shell/plugin-bash-like": "^10.7.6",
-    "@kui-shell/plugin-carbon-themes": "^10.7.6",
-    "@kui-shell/plugin-client-common": "^10.7.6",
-    "@kui-shell/plugin-core-support": "^10.7.6",
-    "@kui-shell/plugin-electron-components": "^10.7.6",
+    "@kui-shell/core": "^10.8.1",
+    "@kui-shell/plugin-bash-like": "^10.8.1",
+    "@kui-shell/plugin-carbon-themes": "^10.8.1",
+    "@kui-shell/plugin-client-common": "^10.8.1",
+    "@kui-shell/plugin-core-support": "^10.8.1",
+    "@kui-shell/plugin-electron-components": "^10.8.1",
     "@kui-shell/plugin-example": "file:./plugins/plugin-example",
-    "@kui-shell/plugin-git": "^10.7.6",
-    "@kui-shell/plugin-kubectl": "^10.7.6",
-    "@kui-shell/plugin-patternfly4-themes": "^10.7.6",
-    "@kui-shell/plugin-proxy-support": "^10.7.6"
+    "@kui-shell/plugin-git": "^10.8.1",
+    "@kui-shell/plugin-kubectl": "^10.8.1",
+    "@kui-shell/plugin-patternfly4-themes": "^10.8.1",
+    "@kui-shell/plugin-proxy-support": "^10.8.1"
   }
 }

--- a/plugins/plugin-client-notebooks/src/autoplay.ts
+++ b/plugins/plugin-client-notebooks/src/autoplay.ts
@@ -1,0 +1,38 @@
+/*
+ * Copyright 2021 The Kubernetes Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @return if found, the autoplay.json config, which will be a list of
+ * notebooks to play on load.
+ *
+ */
+export default function autoplay(): string[] {
+  try {
+    const autoplay = require('@kui-shell/client/config.d/autoplay.json')
+    if (!Array.isArray(autoplay)) {
+      console.error('autoplay config is not an array')
+      return []
+    } else if (autoplay.length > 0 && typeof autoplay[0] !== 'string') {
+      console.error('autoplay config is not an array')
+      return []
+    } else {
+      return autoplay
+    }
+  } catch (err) {
+    // no autoplay...
+    return []
+  }
+}

--- a/plugins/plugin-client-notebooks/src/index.tsx
+++ b/plugins/plugin-client-notebooks/src/index.tsx
@@ -23,8 +23,8 @@ import { CurrentContext, CurrentNamespace } from '@kui-shell/plugin-kubectl/comp
 import { ProxyOfflineIndicator } from '@kui-shell/plugin-proxy-support'
 import { Search } from '@kui-shell/plugin-electron-components'
 
+import auto from './autoplay'
 import { productName } from '@kui-shell/client/config.d/name.json'
-import autoplay from '@kui-shell/client/config.d/autoplay.json'
 
 /**
  * We will set this bit when the user dismisses the Welcome to Kui
@@ -40,6 +40,8 @@ import autoplay from '@kui-shell/client/config.d/autoplay.json'
  *
  */
 export default function renderMain(props: KuiProps) {
+  const autoplay = auto()
+
   return (
     <Kui
       noHelp

--- a/plugins/plugin-client-notebooks/src/preload.ts
+++ b/plugins/plugin-client-notebooks/src/preload.ts
@@ -14,15 +14,18 @@
  * limitations under the License.
  */
 
-import autoplay from '@kui-shell/client/config.d/autoplay.json'
+import auto from './autoplay'
 
 /**
  * Register the welcome notebook
  *
  */
 export default async () => {
-  const { notebookVFS } = await import('@kui-shell/plugin-core-support')
-  notebookVFS.cp(undefined, autoplay.map(notebook => (
-    `plugin://client/notebooks/${notebook}`
-  )), '/kui')
+  const autoplay = auto()
+  if (autoplay.length > 0) {
+    const { notebookVFS } = await import('@kui-shell/plugin-core-support')
+    notebookVFS.cp(undefined, autoplay.map(notebook => (
+      `plugin://client/notebooks/${notebook}`
+    )), '/kui')
+  }
 }

